pkgs/applications/networking/cluster/k3s/README.md

```diff
@@ -1,114 +1,26 @@
-# k3s versions
+# K3s
 
-K3s, Kubernetes, and other clustered software has the property of not being able to update atomically. Most software in nixpkgs, like for example bash, can be updated as part of a "nixos-rebuild switch" without having to worry about the old and the new bash interacting in some way.
+K3s is a simplified version of [Kubernetes](https://wiki.nixos.org/wiki/Kubernetes). It bundles all components for a kubernetes cluster into a few of small binaries.
 
-K3s/Kubernetes, on the other hand, is typically run across several NixOS machines, and each NixOS machine is updated independently. As such, different versions of the package and NixOS module must maintain compatibility with each other through temporary version skew during updates.
+## Usage
 
-The upstream Kubernetes project [documents this in their version-skew policy](https://kubernetes.io/releases/version-skew-policy/#supported-component-upgrade-order).
+Starter usage examples can be found in [docs/USAGE.md](docs/USAGE.md). 
 
-Within nixpkgs, we strive to maintain a valid "upgrade path" that does not run
-afoul of the upstream version skew policy.
+## Configuration Examples
 
-## Upstream release cadence and support
+The following documents help deal with special use cases and real work examples for cluster setup.
 
-K3s is built on top of K8s, and typically provides a similar release cadence and support window (simply by cherry-picking over k8s patches). As such, we assume k3s's support lifecycle is identical to upstream K8s.
+* [ZFS Snapshot Support](docs/examples/ZFS.md)
+* [Nvidia Support](docs/examples/NVIDIA.md)
+* [Storage Examples](docs/examples/STORAGE.md)
 
-This is documented upstream [here](https://kubernetes.io/releases/patch-releases/#support-period).
+## Cluster Maintenance and Troubleshooting
 
-In short, a new Kubernetes version is released roughly every 4 months, and each release is supported for a little over 1 year.
+For cluster maintenance and troubleshooting information, please check out [docs/CLUSTER_UPKEEP.md](docs/CLUSTER_UPKEEP.md).
```

Comment:
```suggestion
```

Reply:
Based on previous comments on comments as headers, thought would be then to just change this to something like: 

```
* [Cluster Upkeep](docs/CLUSTER_UPKEEP.md)
```

Reply:
Feel free to organize things as you wish. I don't want to nitpick. I just wanted to convey to you the idea we should drop as many words as possible. To make things concise. Goal should be to make documentation shorter but meaningful (concise). As I explained in another comment. If we are able to do it, it is going to a nice one.